### PR TITLE
feat(codaveri-evaluation): added support for base64 encoding non UTF-8 files

### PR DESCRIPTION
**Pull request by adi-herwana-nus**
Tested with Python and R examples:
- For Python, tested with reading random bytes from file, and unpacking Python [pickle](https://docs.python.org/3/library/pickle.html) file
- For R, tested with sample `.RData` file.

### CODE REVIEW

File: app/services/course/assessment/question/programming_codaveri/python/python_package_service.rb [hunk 1]
```diff
@@ -101,13 +103,19 @@ def extract_supporting_solution_files
...
-    supporting_solution_object[:type] = 'internal' # 'external' s3 upload not yet implemented by codaveri
-    supporting_solution_object[:path] = filename.to_s
-    supporting_solution_object[:content] = content
+    supporting_file_object = default_codaveri_data_file_template
+
+    supporting_file_object[:type] = 'internal' # 'external' s3 upload not yet implemented by codaveri
+    supporting_file_object[:path] = filename.to_s
+    if content.force_encoding('UTF-8').valid_encoding?
+      supporting_file_object[:content] = content
+      supporting_file_object[:encoding] = 'utf8'
```
Comment (cysjonathan) [inline app/services/course/assessment/question/programming_codaveri/python/python_package_service.rb:112]: check if utf8 or utf-8

File: app/services/course/assessment/question/programming_codaveri/r/r_package_service.rb [hunk 2]
```diff
@@ -102,13 +104,19 @@ def extract_supporting_solution_files
...
+      supporting_file_object[:content] = content
+      supporting_file_object[:encoding] = 'utf8'
+    else
+      supporting_file_object[:content] = Base64.strict_encode64(content)
+      supporting_file_object[:encoding] = 'base64'
+    end
 
-    @data_files.append(supporting_solution_object)
+    @data_files.append(supporting_file_object)
   end
```
Comment (cysjonathan) [inline app/services/course/assessment/question/programming_codaveri/r/r_package_service.rb:120]: Nit: time to consider moving this function out to a common file, given that this is language agnostic, and we'll be having C# and Javascript soon. Okay to wait till when we add those languages.

File: app/services/course/assessment/question/programming_codaveri/python/python_package_service.rb (hunk 1 above)
Comment (adi-herwana-nus) [reply app/services/course/assessment/question/programming_codaveri/python/python_package_service.rb:112]: Codaveri confirmed it to be `utf8` yesterday.

File: app/services/course/assessment/question/programming_codaveri/r/r_package_service.rb (hunk 2 above)
Comment (adi-herwana-nus) [reply app/services/course/assessment/question/programming_codaveri/r/r_package_service.rb:120]: I had the same thought, but still thinking over the code structure after doing that. We will definitely do so when we add the C# and JavaScript Codaveri services.

**Review by cysjonathan — APPROVED** [review]

**Pull request merged**